Implement middle-clicking to invert a gradient in the Gradient editor

Calinou opened the pull request:
This can make it easier to work on more complex gradients.

PS: Is there a more suited shortcut that could be used for this? I haven't really looked how gradient editors in other software handle this feature.

groud commented:
Couldn't we have a popup menu on right clic instead ? Using middle click looks weird.

Calinou commented:
@groud Right-click is currently used to delete gradient stops, so I'm not sure how well it would work alongside.

akien-mga commented:
> I noticed the Curve editor uses middle-click for removing points, and right-click to display a contextual menu. Maybe we could do the same in the Gradient editor?

I think that would be good, yes.

aaronfranke commented:
@Calinou Is this still desired? If so, it needs to be rebased on the latest master branch.

Calinou commented:
@aaronfranke Yes, but it needs to be remade to use a context menu instead.

Calinou committed:
Add a context menu to the Gradient editor to invert a gradient's points

Removing points is now done by middle-clicking instead of right-clicking,
like in the Curve editor.

Calinou commented:
I amended the pull request to use a context menu instead

Calinou commented on scene/gui/gradient_edit.cpp:
```diff
@@ -126,8 +132,15 @@ void GradientEdit::_gui_input(const Ref<InputEvent> &p_event) {
-	if (mb.is_valid() && mb->get_button_index() == 1 && mb->is_pressed() && mb->get_alt()) {
+	// Show context menu on right click.
+	if (mb.is_valid() && mb->get_button_index() == BUTTON_RIGHT && mb->is_pressed()) {
+		context_menu->set_position(get_global_transform().xform(mb->get_position()));
```
Unfortuntely, the context menu doesn't spawn at the right position. I don't know why, as I used the same method in https://github.com/godotengine/godot/pull/38593 and it works there.

EricEzaM commented:
Middle clicking is not very discoverable for the end user... A right click context menu is better, but still not great. We are not short on display real estate, why not just add it as a small button to the side of the gradient display?

Calinou commented:
@EricEzaM I find the gradient editor pretty narrow as it is already. I'd rather not add a button for something you don't need very often besides the gradient.

Calinou commented:
This is now implemented in 4.0.alpha by https://github.com/godotengine/godot/pull/53321, closing.

![image](https://user-images.githubusercontent.com/180032/174464660-4dd85f78-17ef-4342-a660-d46216128b8e.png)

https://github.com/godotengine/godot/pull/53321 could likely be backported to `3.x` in a backwards-compatible manner, but it may not be trivial.
